addons/Context/Tweek.Drivers.Context.InMemory/InMemoryContext.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System.Collections.Concurrent;
+using FSharpUtils.Newtonsoft;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using System.Threading.Tasks;
+using Tweek.Engine.DataTypes;
+using Tweek.Engine.Drivers.Context;
+
+namespace Tweek.Drivers.Context.InMemory
+{
+    public class InMemoryContext : IContextDriver
+    {
+        private ConcurrentDictionary<string, Dictionary<string, JsonValue>> _data;
+
+        public InMemoryContext()
+        {
+            _data = new ConcurrentDictionary<string, Dictionary<string, JsonValue>>();
+        }
+
+        private static string GetKey(Identity identity) => $"identity_{identity.Type}_{identity.Id}";
+
+        public Task RemoveFromContext(Identity identity, string key)
+        {
+            if (_data.TryGetValue(GetKey(identity), out var context)) context.Remove(key);
+            return Task.CompletedTask;
+        }
+
+        public Task DeleteContext(Identity identity)
+        {
+            _data.Remove(GetKey(identity), out _);
+            return Task.CompletedTask;
+        }
+
+        public Task AppendContext(Identity identity, Dictionary<string, JsonValue> context)
+        {
+            var key = GetKey(identity);
+
+            if (_data.TryGetValue(key, out var existingContext))
+            {
+                _data[key] = MergeContexts(existingContext, context);
+                return Task.CompletedTask;
+            }
+
+            var contextWithDate = new Dictionary<string, JsonValue>(context);
+            contextWithDate.Add("@CreationDate", JsonValue.NewString(DateTimeOffset.UtcNow.ToString()));
+
+            _data[key] = contextWithDate;
+            return Task.CompletedTask;
```

Comment:
Maybe to use `AddOrUpdate` method as described [here](https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.addorupdate?view=netstandard-2.1)